### DISCUSSION TITLE
Webhooks

### DISCUSSION THREAD

**Discussion by scottf-tvw** [bluenviron/mediamtx#2323] [General]
A solid need for this feature exists.  I am sorry but a [SIGINT wait in a script](https://github.com/bluenviron/mediamtx/discussions/554#discussioncomment-1282935) is kludgy and definitely not ideal.  I was about to make the changes necessary to see a webhook in the context of a source connecting and disconnecting (publish/unpublish), until I saw the PR  (https://github.com/bluenviron/mediamtx/pull/2207) and someone had already done it. Sadly I saw it was not merged and closed. There is no public discussion on why that decision was bad for this project other than you told the author the reasons why, but not publicly. I am just trying to understand your perspective.  [aler9](https://github.com/aler9) Can you take a minute to explain why this PR was denied?  I am not advocating for anyone, it's just that there seemed to be a decent solution for webhook call outs for when a stream source connects and disconnects from publishing.  There seems no better solution other than [this](https://github.com/bluenviron/mediamtx/issues/1464) ticket for an enhancement.  I certainly don't want to put the work in to contribute to just get it shot down for lack of knowledge of the reasons why this other PR was not merged. 
Thanks for your time

**Comment by aler9** [comment]
Hello, i perfectly agree with you on the fact that the integration interface (we're talking about this) has to be simplified, but at the same time you have to understand that in any project, any additional feature means additional maintenance work, especially if such feature is just a simplification of an existing one, that will still exist, causing confusion in users about which one to use.

Webhooks can already be used without issues by placing `curl` or `wget` inside `runOn*`.

runOnStop hooks are planned, are tracked in #1464 and will be surely developed in the future (and a PR would be welcome).

I'd rather avoid merging specific integrations (i.e. events sent to Redis, Kafka, MQTT, AMQP) since they would open the way for any kind of integration, tripling the dependencies, while the same exact thing can be done by placing a command line in `runOn*` with a minimal overhead.

**Reply by scottf-tvw** [reply]
Thank you for the explanation and I agree that this can lead to a maintainers nightmare. I look forward to `runOnStop` and possibly other raised events being a part of this project.

[a comment by an automated account is not shown]
